x-pack/plugin/ent-search/src/main/java/module-info.java

```diff
@@ -0,0 +1,15 @@
+module org.elasticsearch.application {
+    requires org.apache.lucene.core;
+
+    requires org.elasticsearch.base;
+    requires org.elasticsearch.logging;
+    requires org.elasticsearch.server;
+    requires org.elasticsearch.xcontent;
+    requires org.elasticsearch.xcore;
+
+    exports org.elasticsearch.xpack.application.analytics;
+    exports org.elasticsearch.xpack.application.analytics.action;
+
+    exports org.elasticsearch.xpack.application.search;
+    exports org.elasticsearch.xpack.application.search.action;
```

Comment:
We could be more strict on which module can use our code but I do not see a huge benefits in doing it like in security.